### PR TITLE
DDP-4911 support test result dsm notification

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

A handful of new features:

* __Activity content substitutions:__ added participant_guid and various test result substitutions. And timezone is necessary to render the test_result_time_completed appropriately for the user.
* __Activity status summary:__ the summary text associated with each activity status can now be rendered using activity content substitution values, therefore allowing us to put in the test_result_time_completed value.
* __DSM notification:__ added `TEST_RESULT` type so we can create events triggered based on that. It also allows us to narrow down the trigger to a specific test result type, if we need that.
* __Instance creation:__ this event action will now add in test result data as activity content substitution values to the newly created activity instance.
* __PEX:__ new syntax to query for snapshotted activity content substitution values.

Also added the activity definition for TestBoston's `RESULT_REPORT` activity. Still waiting on the "other" result report.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.

## Testing

- [x] I have written automated positive tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality

## Release

- [x] These changes require no special release procedures--just code!

### CODE REVIEW

File: pepper-apis/src/main/antlr4/org/broadinstitute/ddp/pex/lang/Pex.g4 [hunk 1]
```diff
@@ -22,6 +22,7 @@ expr
 query
   : 'user' '.' study '.' studyPredicate                                                  # StudyQuery
   | 'user' '.' study '.' form '.' formPredicate                                          # FormQuery
+  | 'user' '.' study '.' form '.' instance '.' formInstancePredicate                     # FormInstanceQuery
```
Comment (yufengwng) [inline pepper-apis/src/main/antlr4/org/broadinstitute/ddp/pex/lang/Pex.g4:25]: We need to narrow do to a particular activity instance because content substitution values are associated with an activity instance.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/json/dsm/DsmNotificationPayload.java [hunk 2]
```diff
@@ -19,6 +20,9 @@
     @SerializedName("eventDate")
     private long eventDate;
 
+    @SerializedName("eventData")
+    private JsonElement eventData;
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/json/dsm/DsmNotificationPayload.java:24]: Using `JsonElement` for flexibility since other event types might have a different structure for their data.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/model/dsm/TestResult.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+public class TestResult {
+
+    @NotBlank
+    @SerializedName("result")
+    private String result;
+
+    @NotBlank
+    @SerializedName("reason")
+    private String reason;
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/model/dsm/TestResult.java]: Can get rid of `reason` if the "other" report does not use it.

File: study-builder/studies/testboston/i18n/es.conf [hunk 4]
```diff
@@ -102,7 +102,7 @@
...
     "s3_injured_p2": "En las investigaciones, a veces se producen lesiones aunque nadie tenga la culpa. No planeamos pagarle ni darle ningún tipo de indemnización en el caso de que sufra alguna lesión. Sin embargo, al firmar este formulario no renuncia a ninguno de los derechos que le otorga la ley.",
-    "s3_injured_p2": "Si cree que ha sufrido una lesión o tiene algún problema de salud por participar en este estudio de investigación, informe cuanto antes a la persona encargada del estudio. El nombre y el número de teléfono de la investigadora figuran al inicio de este formulario de consentimiento.",
+    "s3_injured_p3": "Si cree que ha sufrido una lesión o tiene algún problema de salud por participar en este estudio de investigación, informe cuanto antes a la persona encargada del estudio. El nombre y el número de teléfono de la investigadora figuran al inicio de este formulario de consentimiento.",
```
Comment (yufengwng) [inline study-builder/studies/testboston/i18n/es.conf:105]: I introduced a typo, sorry about that, fixing it now.

File: study-builder/studies/testboston/study-events.conf [hunk 5]
```diff
@@ -179,7 +204,7 @@
         "activityCode": ${id.act.longitudinal}
       },
       "cancelExpr": """!user.studies["testboston"].forms["BASELINE_SYMPTOM"].hasInstance()""",
-      "dispatchToHousekeeping": true,
+      "dispatchToHousekeeping": false,
```
Comment (yufengwng) [inline study-builder/studies/testboston/study-events.conf:207]: These two events does not need to be dispatched to Housekeeping.

**Comment by zyme** [comment]
Something we  should  expect--accidental duplicate results from CE.   Can we ensure that we only send a notification for the  unique combination of patient  guid + collection time + result time?  Not sure if that filter should  go here, in DSM,  or in  Birch.  CE has accidentally sent multiple of  the  same emails to patients in the past, so lets assume we'll see that again.

File: study-builder/studies/testboston/result-report.conf [hunk 6]
```diff
@@ -0,0 +1,412 @@
...
+                "translations": [
+                  { "language": "en", "text": ${i18n.en.result_report.pos_item4} },
+                  { "language": "es", "text": ${i18n.es.result_report.pos_item4} },
+                  { "language": "ht", "text": ${i18n.ht.result_report.pos_item4} }
+                ]
+              }
+            ]
+          },
+          "blockType": "CONTENT"
+          "shownExpr": """user.studies["testboston"].forms["RESULT_REPORT"].instances[specific].snapshotSubstitution("DDP_TEST_RESULT_CODE") == "POSITIVE""""
```
Comment (MocanaAtBroad) [inline study-builder/studies/testboston/result-report.conf:148]: specific?

File: study-builder/studies/study-example.conf [hunk 7]
```diff
@@ -500,7 +500,10 @@
     {
       "trigger": {
         "type": "DSM_NOTIFICATION",
-        "dsmEvent": "SALIVA_RECEIVED | BLOOD_RECEIVED | BLOOD_SENT | BLOOD_SENT_4WK | TESTBOSTON_SENT | TESTBOSTON_DELIVERED | TESTBOSTON_RECEIVED"
+        "dsmEvent": """SALIVA_RECEIVED | BLOOD_RECEIVED | BLOOD_SENT | BLOOD_SENT_4WK
+                  | TESTBOSTON_SENT | TESTBOSTON_DELIVERED | TESTBOSTON_RECEIVED | TEST_RESULT""",
```
Comment (MocanaAtBroad) [inline study-builder/studies/study-example.conf]: These events need to be TESTBOSTON-specific? kits sent/delivered/received and test result available sound pretty generic and reusable to me.

File: pepper-apis/src/main/resources/db-changes/schema/DDP-4911-test-result-event.xml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        <insert tableName="test_result_event_type">
+            <column name="test_result_event_type_code" value="POSITIVE"/>
+        </insert>
+        <insert tableName="test_result_event_type">
+            <column name="test_result_event_type_code" value="OTHER"/>
+        </insert>
+        <insert tableName="test_result_event_type">
+            <column name="test_result_event_type_code" value="ANY"/>
+        </insert>
+    </changeSet>
```
Comment (MocanaAtBroad) [inline pepper-apis/src/main/resources/db-changes/schema/DDP-4911-test-result-event.xml]: Do we want to encode a test result as an event type? Would it just not be sufficient to know that a test result is available?

**Comment by yufengwng** [comment]
@zyme won't accidental duplicate results come from GP/pipelines instead of CE/Ellkay? Birch does try to only send result once, but DSM should have de-duplication in place in case of pubsub "at-least-once" delivery. I'm aware that in the past CE did resend _almost_ duplicate orders (e.g. same order sample id but different timestamp or something) -- I recall that the older order was cancelled/unsat-ed and the result was only delivered by GP for the newer order. But that scenario also concerns me regarding what the participant will see... CC @pegahtah

File: pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserActivityInstanceListRoute.java [hunk 9]
```diff
@@ -50,18 +54,46 @@ public UserActivityInstanceListRoute(ActivityInstanceDao activityInstanceDao) {
...
+                    }
+                    Map<String, String> subs = substitutions.getOrDefault(summary.getActivityInstanceId(), new HashMap<>());
+                    var provider = new RenderValueProvider.Builder()
+                            .withSnapshot(sharedSnapshot)
+                            .withSnapshot(subs)
+                            .build();
+                    Map<String, Object> context = new HashMap<>();
+                    context.put(I18nTemplateConstants.DDP, provider);
+                    String summaryText = renderer.renderToString(summary.getActivitySummary(), context);
+                    summary.setActivitySummary(summaryText);
```
Comment (MocanaAtBroad) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserActivityInstanceListRoute.java]: This looks a little too low-level to be part of the route. Can this be pushed down to Dao or some other intermediate service. TMI.

File: study-builder/studies/testboston/i18n/en.conf [hunk 10]
```diff
@@ -168,6 +168,36 @@
...
+    "label_id": "TestBoston Study ID:",
+    "label_dob": "Date of Birth:",
+    "label_date": "SARS-CoV-2 PCR Test Date:",
+    "label_result": "SARS-CoV-2 PCR Test Result:",
+    "info": "Important Information Related to your Test Results:",
+    "pos_item1": "This result means that SARS-CoV-2, the virus that causes COVID-19, was <strong>detected</strong> in your nasal swab sample.",
+    "pos_item2": "<strong>This is a research test result and requires clinical confirmation of COVID-19 infection.</strong> The test was conducted as a part of a research study of home self-testing in the Clinical Research Sequencing Platform lab at the Broad Institute.",
+    "pos_item3": "Please contact your primary care provider's office with additional health questions. If you do not have a PCP or are unable to reach them, you can also call the Partners HealthCare COVID-19 hotline at 617-724-7000.",
+    "pos_item4": "If you are a Mass General Brigham employee, please call 617-724-8100 for tailored guidance on confirmatory testing and when you are permitted to return to work. If you work in health care, please call your employer's Occupational Health Department.",
```
Comment (fat23cat) [inline study-builder/studies/testboston/i18n/en.conf]:
```suggestion
    "pos_item3": "Please contact your primary care provider's office with additional health questions. If you do not have a PCP or are unable to reach them, you can also call the Partners HealthCare COVID-19 hotline at <a class=\"Link Link_semi-bold\" href=\"tel:617-724-7000\">617-724-7000</a>.",
    "pos_item4": "If you are a Mass General Brigham employee, please call <a class=\"Link Link_semi-bold\" href=\"tel:617-724-8100\">617-724-8100</a> for tailored guidance on confirmatory testing and when you are permitted to return to work. If you work in health care, please call your employer's Occupational Health Department.",

```

File: study-builder/studies/testboston/i18n/en.conf [hunk 11]
```diff
@@ -168,6 +168,36 @@
...
+    "recommend": "We recommend that you to obtain a clinical test to confirm the result of the at-home test.",
+    "neg_item1": "This result means that SARS-CoV-2, the virus that causes COVID-19, was <strong>not detected</strong> in your nasal swab sample.",
+    "neg_item2": "<strong>This is a research result and should be confirmed with a clinically validated COVID-19 test.</strong> The test was conducted as a part of a research study of home self-testing in the Clinical Research Sequencing Platform lab at the Broad Institute.",
+    "neg_item3": "If you develop any new symptoms of COVID 19 infection such as fever, sore throat, new cough, new runny nose, muscle aches, loss of smell or taste or trouble breathing you should contact your PCP's office. If you do not have a PCP or are unable to reach them, you can call the Partners Hotline at 617-724-7000.",
+    "neg_item4": "You may also always contact the team at (617)-525-4220 or at info@testboston.org for additional information.",
```
Comment (fat23cat) [inline study-builder/studies/testboston/i18n/en.conf]:
```suggestion
    "neg_item3": "If you develop any new symptoms of COVID 19 infection such as fever, sore throat, new cough, new runny nose, muscle aches, loss of smell or taste or trouble breathing you should contact your PCP's office. If you do not have a PCP or are unable to reach them, you can call the Partners Hotline at <a class=\"Link Link_semi-bold\" href=\"tel:617-724-7000\">617-724-7000</a>.",
    "neg_item4": "You may also always contact the team at <a class=\"Link Link_semi-bold\" href=\"tel:(617)-525-4220\">(617)-525-4220</a> or at <a class=\"Link Link_semi-bold\" href=\"mailto:info@testboston.org\">info@testboston.org</a> for additional information.",

```

File: study-builder/studies/testboston/i18n/en.conf [hunk 12]
```diff
@@ -168,6 +168,36 @@
...
+    "more_info_item2_pos": "<strong>Boston residents:</strong> Find reliable information including information on free testing at the Boston Public Health Commission, (617) 534-5395. Also check the website: <a href=\"https://www.bphc.org/onlinenewsroom/Blog/Lists/Posts/Post.aspx?List=24ee0d58%2D2a85%2D4a4a%2D855b%2Df5af9d781627&ID=1282&RootFolder=%2Fonlinenewsroom%2FBlog%2FLists%2FPosts&Source=https%3A%2F%2Fwww%2Ebphc%2Eorg%2FPages%2Fdefault%2Easpx&Web=03126e14%2D4972%2D4333%2Db8a3%2D800cbc1cafce\" target=\"_blank\">BPHC</a>.",
+    "more_info_item2_neg": "<strong>Boston residents:</strong> Find reliable information at the Boston Public Health Commission, (617) 534-5395. Also check the website: <a href=\"https://www.bphc.org/onlinenewsroom/Blog/Lists/Posts/Post.aspx?List=24ee0d58%2D2a85%2D4a4a%2D855b%2Df5af9d781627&ID=1282&RootFolder=%2Fonlinenewsroom%2FBlog%2FLists%2FPosts&Source=https%3A%2F%2Fwww%2Ebphc%2Eorg%2FPages%2Fdefault%2Easpx&Web=03126e14%2D4972%2D4333%2Db8a3%2D800cbc1cafce\" target=\"_blank\">BPHC</a>.",
```
Comment (fat23cat) [inline study-builder/studies/testboston/i18n/en.conf]:
```suggestion
    "more_info_item1": "<strong>Massachusetts residents:</strong> The Massachusetts Department of Public Health (DPH) Hotline can be reached at <a class=\"Link Link_semi-bold\" href=\"tel:(617) 983-6800\">(617) 983-6800</a> or at <a class=\"Link Link_semi-bold\" href=\"tel:211\">211</a>, 24 hours a day, 7 days per week. <strong>TTY/TDD</strong> <a class=\"Link Link_semi-bold\" href=\"tel:(617) 624-6001\">(617) 624-6001</a>. Also check the website: <a href=\"https://www.mass.gov/resource/information-on-the-outbreak-of-coronavirus-disease-2019-covid-19\" target=\"_blank\">Massachusetts DPH</a>.",
    "more_info_item2_pos": "<strong>Boston residents:</strong> Find reliable information including information on free testing at the Boston Public Health Commission, <a class=\"Link Link_semi-bold\" href=\"tel:(617) 534-5395\">(617) 534-5395</a>. Also check the website: <a href=\"https://www.bphc.org/onlinenewsroom/Blog/Lists/Posts/Post.aspx?List=24ee0d58%2D2a85%2D4a4a%2D855b%2Df5af9d781627&ID=1282&RootFolder=%2Fonlinenewsroom%2FBlog%2FLists%2FPosts&Source=https%3A%2F%2Fwww%2Ebphc%2Eorg%2FPages%2Fdefault%2Easpx&Web=03126e14%2D4972%2D4333%2Db8a3%2D800cbc1cafce\" target=\"_blank\">BPHC</a>.",
    "more_info_item2_neg": "<strong>Boston residents:</strong> Find reliable information at the Boston Public Health Commission, <a class=\"Link Link_semi-bold\" href=\"tel:(617) 534-5395\">(617) 534-5395</a>. Also check the website: <a href=\"https://www.bphc.org/onlinenewsroom/Blog/Lists/Posts/Post.aspx?List=24ee0d58%2D2a85%2D4a4a%2D855b%2Df5af9d781627&ID=1282&RootFolder=%2Fonlinenewsroom%2FBlog%2FLists%2FPosts&Source=https%3A%2F%2Fwww%2Ebphc%2Eorg%2FPages%2Fdefault%2Easpx&Web=03126e14%2D4972%2D4333%2Db8a3%2D800cbc1cafce\" target=\"_blank\">BPHC</a>.",

```

File: study-builder/studies/testboston/i18n/en.conf [hunk 13]
```diff
@@ -168,6 +168,36 @@
...
+    "other_item": "This result means we are unable to get a result from your nasal swab. Some reasons this may occur include:",
+    "other_reason1": "A problem with the sample received. This can occur if the swab was not in the tube, the swab was found upside down or if the tube was not labelled correctly.",
+    "other_reason2": "The swab didnot arrive in the lab within 72 hours of being collected.",
+    "other_reason3": "The swab failed lab quality procedures.",
+    "other_reason4": "The sample was damaged or lost during testing.",
+    "other_contact": "You may also always contact the team at info@testboston.org for additional information or call the TestBoston study team at (617) 525-4220 if you would like  an additional test kit sent to re-do the anterior nasal swab collection and test.",
```
Comment (fat23cat) [inline study-builder/studies/testboston/i18n/en.conf]:
```suggestion
    "other_contact": "You may also always contact the team at <a class=\"Link Link_semi-bold\" href=\"mailto:info@testboston.org\">info@testboston.org</a> for additional information or call the TestBoston study team at <a class=\"Link Link_semi-bold\" href=\"tel:(617) 525-4220\">(617) 525-4220</a> if you would like  an additional test kit sent to re-do the anterior nasal swab collection and test.",
```

File: study-builder/studies/testboston/i18n/en.conf [hunk 14]
```diff
@@ -168,6 +168,36 @@
...
+    "more_info_item2_neg": "<strong>Boston residents:</strong> Find reliable information at the Boston Public Health Commission, (617) 534-5395. Also check the website: <a href=\"https://www.bphc.org/onlinenewsroom/Blog/Lists/Posts/Post.aspx?List=24ee0d58%2D2a85%2D4a4a%2D855b%2Df5af9d781627&ID=1282&RootFolder=%2Fonlinenewsroom%2FBlog%2FLists%2FPosts&Source=https%3A%2F%2Fwww%2Ebphc%2Eorg%2FPages%2Fdefault%2Easpx&Web=03126e14%2D4972%2D4333%2Db8a3%2D800cbc1cafce\" target=\"_blank\">BPHC</a>.",
+    "other_item": "This result means we are unable to get a result from your nasal swab. Some reasons this may occur include:",
+    "other_reason1": "A problem with the sample received. This can occur if the swab was not in the tube, the swab was found upside down or if the tube was not labelled correctly.",
+    "other_reason2": "The swab didnot arrive in the lab within 72 hours of being collected.",
```
Comment (fat23cat) [inline study-builder/studies/testboston/i18n/en.conf]: Typo?
```suggestion
    "other_reason2": "The swab did not arrive in the lab within 72 hours of being collected.",
```

**Comment by zyme** [comment]
@yufengwng thanks for the explanation on duplicates--you're right, this is all in CE, and since we get our results from GP, we shouldn't run into this issue directly via CE.

However, we have seen cases where we have had to re-publish results for Ellkay.  So far we have sent these from Birch, but I have some concern about us needing to do some kind of administrative clean-up by re-pushing results from GP instead of Birch.  If we see `corrected` or `revised`, that should always go through, but if we see > 1 result for the same patient guid, kit, and collection time, we should flag that as an error and not notify the participant.

File: study-builder/studies/testboston/result-report.conf (hunk 6 above)
Comment (yufengwng) [reply study-builder/studies/testboston/result-report.conf:148]: There will be multiple instances of this activity. Each instance will have it's own set of substitutions. Therefore, the PEX need to look at a _specific_ activity instance, the one that we're dealing with at runtime.

File: study-builder/studies/study-example.conf (hunk 7 above)
Comment (yufengwng) [reply study-builder/studies/study-example.conf]: I recall that the `TESTBOSTON_*` kit events are specific to TestBoston because it's a different physical kit than saliva or blood. I concur that the test result is more general, thus I named it `TEST_RESULT`.

File: pepper-apis/src/main/resources/db-changes/schema/DDP-4911-test-result-event.xml (hunk 8 above)
Comment (yufengwng) [reply pepper-apis/src/main/resources/db-changes/schema/DDP-4911-test-result-event.xml]: Not sure. Some studies might be interested in that. For TestBoston it's sufficient to know that a test result is available because our implementation for TestBoston is to use one single activity.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserActivityInstanceListRoute.java (hunk 9 above)
Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserActivityInstanceListRoute.java]: I'll investigate and refactor.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/model/dsm/TestResult.java (hunk 3 above)
Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/model/dsm/TestResult.java]: We don't need `reason` anymore.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserActivityInstanceListRoute.java (hunk 9 above)
Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserActivityInstanceListRoute.java]: Do you think we can keep this here? I think it's a bit more efficient since it's this is done after we have filtered down the list of activity instances, and I don't think any callers of the DAO really cares about the summary text?

Comment (MocanaAtBroad) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserActivityInstanceListRoute.java]: I'll go along with your assessment. Just tend to be suspicious of doing much work on routes.

Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserActivityInstanceListRoute.java]: I have encapsulated this functionality in the DAO so other callers can use it if needed.

**Review by fat23cat — COMMENTED** [review]
The translation part(links and phones) looks good!

File: study-builder/studies/testboston/i18n/en.conf [hunk 15]
```diff
@@ -168,6 +168,36 @@
...
+    "more_info_item2_neg": "<strong>Boston residents:</strong> Find reliable information at the Boston Public Health Commission, <a class=\"Link Link_semi-bold\" href=\"tel:(617) 534-5395\">(617) 534-5395</a>. Also check the website: <a href=\"https://www.bphc.org/onlinenewsroom/Blog/Lists/Posts/Post.aspx?List=24ee0d58%2D2a85%2D4a4a%2D855b%2Df5af9d781627&ID=1282&RootFolder=%2Fonlinenewsroom%2FBlog%2FLists%2FPosts&Source=https%3A%2F%2Fwww%2Ebphc%2Eorg%2FPages%2Fdefault%2Easpx&Web=03126e14%2D4972%2D4333%2Db8a3%2D800cbc1cafce\" target=\"_blank\">BPHC</a>.",
+    "other_item": "This result means we are unable to get a result from your nasal swab. Some reasons this may occur include:",
+    "other_reason1": "A problem with the sample received. This can occur if the swab was not in the tube, the swab was found upside down or if the tube was not labelled correctly.",
+    "other_reason2": "The swab did not arrive in the lab within 72 hours of being collected.",
```
Comment (ssettipalli) [inline study-builder/studies/testboston/i18n/en.conf:196]: As per zims comment: https://broadinstitute.slack.com/archives/G012Y2S45KN/p1599748272417200 , we need this text changed to "The sample could not be processed within 72 hours of collection"

File: pepper-apis/src/main/resources/db-changes/schema/DDP-4911-test-result-event.xml (hunk 8 above)
Comment (MocanaAtBroad) [reply pepper-apis/src/main/resources/db-changes/schema/DDP-4911-test-result-event.xml]: Actually, I was focusing more on the idea of specifying that at test result was POSITIVE or NEGATIVE as an even type. This seems more like content of the test result.

Comment (yufengwng) [reply pepper-apis/src/main/resources/db-changes/schema/DDP-4911-test-result-event.xml]: I felt we had to key it off of the "content of the test result" like this because of the nature of the problem. We had 3 different report letters for each of the positive/negative/other outcome. So if we were to create different activities for positive/negative/other then I don't see a way to do it unless you actually look at the content of the test result like this. However, we were able to achieve having just one single activity, so I can remove this now.

File: study-builder/studies/testboston/i18n/en.conf (hunk 15 above)
Comment (yufengwng) [reply study-builder/studies/testboston/i18n/en.conf:196]: In another Slack channel with the PIs, the PIs indicated that we should keep it for now.

File: pepper-apis/src/main/resources/db-changes/schema/DDP-4911-test-result-event.xml (hunk 8 above)
Comment (yufengwng) [reply pepper-apis/src/main/resources/db-changes/schema/DDP-4911-test-result-event.xml]: Removed!

**Review by MocanaAtBroad — APPROVED** [review]
